File: tester/harmony/react_native_modules/reanimated/src/main/ets/ReanimatedPackage.ts
import {RNPackage, TurboModuleContext, TurboModulesFactory} from 'rnoh/ts';
import type {TurboModule} from 'rnoh/ts';
import {ReanimatedModule} from './ReanimatedModule';
class ReanimatedTurboModulesFactory extends TurboModulesFactory {
  createTurboModule(name: string): TurboModule | null {
    if (name === ReanimatedModule.NAME) {
      return new ReanimatedModule(this.ctx);
    }
    return null;
  }
  hasTurboModule(name: string): boolean {
    return name === 'ReanimatedModule';
  }
}
export class ReanimatedPackage extends RNPackage {
  createTurboModulesFactory(ctx: TurboModuleContext): TurboModulesFactory {
    return new ReanimatedTurboModulesFactory(ctx);
  }
}

File: tester/harmony/reanimated/src/main/cpp/ReanimatedPackage.cpp
#include "ReanimatedPackage.h"
#include "ReanimatedModule.h"

using namespace rnoh;
using namespace facebook;

class ReanimatedTurboModuleFactoryDelegate : public TurboModuleFactoryDelegate {
public:
    SharedTurboModule createTurboModule(Context ctx, const std::string &name) const override {
        if (name == "ReanimatedModule") {
            return std::make_shared<ReanimatedModule>(ctx, name);
        }
        return nullptr;
    };
};

std::unique_ptr<TurboModuleFactoryDelegate> ReanimatedPackage::createTurboModuleFactoryDelegate() {
    return std::make_unique<ReanimatedTurboModuleFactoryDelegate>();
}
